.github/workflows/pkgci_regression_test_cpu.yml

```diff
@@ -17,6 +17,9 @@ on:
         type: string
         default: ""
 
+permissions:
+  contents: read
```

Comment:
https://github.com/openxla/iree/actions/runs/8252078447

```
Invalid workflow file: .github/workflows/pkgci.yml#L40
The workflow is not valid. .github/workflows/pkgci.yml (Line: 40, Col: 3): Error calling workflow 'openxla/iree/.github/workflows/pkgci_regression_test_cpu.yml@46d4dcc2c4f6e80194277659fb1fd68f8ce04d01'. The nested job 'linux_x86_64' is requesting 'actions: read', but is only allowed 'actions: none'.
```

Reply:
I explicitly added `actions: read` to for the nested workflows in `pkgci.yml` now, but I am not sure if we really need this.

I was assuming that the default permissions would reflect `Default access (permissive)`, see [Permissions for the GITHUB_TOKEN](https://docs.github.com/en/actions/security-guides/automatic-token-authentication#permissions-for-the-github_token), and I read the docs as `actions: write` would be needed. However, even with permissions set to `Default access (restricted)`, we might not even need to grant `actions: write` permissions. Seems the [Get an artifact](https://docs.github.com/en/rest/actions/artifacts?apiVersion=2022-11-28#get-an-artifact) endpoint can be used "without authentication or the aforementioned permissions if only public resources are requested." which seems to be the case here (not sure about nested workflows). WDYT?

Reply:
Reading through https://docs.github.com/en/actions/using-workflows/reusing-workflows#supported-keywords-for-jobs-that-call-a-reusable-workflow, we might also need to set `contents: read` here:

```diff
  regression_test_cpu:
    name: Regression Test CPU
    permissions:
      actions: read
+     contents: read
    needs: [setup, build_packages]
    if: contains(fromJson(needs.setup.outputs.enabled-jobs), 'regression_test_cpu')
    uses: ./.github/workflows/pkgci_regression_test_cpu.yml
```

Take a look at the latest logs: https://github.com/openxla/iree/actions/runs/8260576118. Weird that those don't even show up on the PR landing page and it shows as "All checks have passed". If you go into https://github.com/openxla/iree/pull/16747/checks then click "PkgCI", you should see similar logs after pushing a commit.

BTW, if you want to iterate on that without triggering other CI jobs, you can add a git trailer to the end of the PR description with this text: `ci-exactly: build_packages,regression_test_amdgpu_rocm,regression_test_amdgpu_vulkan,regression_test_cpu,test_tensorflow_cpu` (as this PR doesn't affect any other workflows)

Reply:
> Reading through https://docs.github.com/en/actions/using-workflows/reusing-workflows#supported-keywords-for-jobs-that-call-a-reusable-workflow, we might also need to set `contents: read` here:
> 
> ```diff
>   regression_test_cpu:
>     name: Regression Test CPU
>     permissions:
>       actions: read
> +     contents: read
>     needs: [setup, build_packages]
>     if: contains(fromJson(needs.setup.outputs.enabled-jobs), 'regression_test_cpu')
>     uses: ./.github/workflows/pkgci_regression_test_cpu.yml
> ```

Since `contents: read` is set at top level it should apply to all jobs in that workflow and I assumed also to nested workflows. Looking at the error we're getting (`The workflow is requesting 'contents: read', but is only allowed 'contents: none'.`), it seems this assumption was wrong -.- Try to find time to look into this tomorrow.